src/main/java/org/opentripplanner/model/plan/Itinerary.java

```diff
@@ -162,13 +179,28 @@ public boolean isOnStreetAllTheWay() {
     return isStreetOnly();
   }
 
+  /**
+   * Returns true if this itinerary has only flex and walking legs.
+   */
+  public boolean isDirectFlex() {
+    return legs.stream().allMatch(l -> l.isFlexibleTrip() || l.isWalkingLeg());
+  }
```

Comment:
The method name indicates that the itinerary must have flex but in this check, flex is optional.

Reply:
OMG, you're right. Walk-only would also return true! That's fixed in https://github.com/opentripplanner/OpenTripPlanner/pull/6050/commits/aa153b773955ada20efea14bff747d209ef453a2